src/chain/shared_state.rs

```diff
@@ -520,15 +520,21 @@ impl SectionProofBlock {
 }
 
 #[derive(Debug, Eq, PartialEq, Clone, Hash, Serialize, Deserialize)]
-pub struct SectionProofChain {
-    genesis_key_info: SectionKeyInfo,
+pub struct SliceProofChain {
```

Comment:
Minor: this is a "slice of SectionProofChain", so I think `SectionProofChainSlice` or just `SectionProofSlice` would be more appropriate name.